File: examples/tcp_probe.nim
import std/net

# OpenClaw TCP port probe (Nim) — checks gateway nodes
proc probe(host: string, port: int): bool =
  var socket = newSocket()
  try:
    socket.connect(host, Port(port), timeout = 3000)
    result = true
  except OSError, TimeoutError:
    result = false
  finally:
    socket.close()

let nodes = [("localhost", 8080), ("localhost", 8081)]
for (host, port) in nodes:
  echo (if probe(host, port): "OK  " else: "FAIL"), " ", host, ":", port

File: examples/tcp_probe.clj
(ns tcp-probe
  (:import [java.net Socket InetSocketAddress]))

;; OpenClaw TCP port probe (Clojure) — checks gateway nodes
(defn probe [host port timeout-ms]
  (try
    (doto (Socket.)
      (.connect (InetSocketAddress. ^String host (int port)) (int timeout-ms))
      (.close))
    true
    (catch Exception _ false)))

(defn -main [& _]
  (doseq [[host port] [["localhost" 8080] ["localhost" 8081]]]
    (println (str (if (probe host port 3000) "OK  " "FAIL") " " host ":" port))))

(-main)
